go/cmd/dolt/commands/sql.go

```diff
@@ -745,8 +778,8 @@ func processQuery(ctx *sql.Context, query string, se *sqlEngine) (sql.Schema, sq
 	case *sqlparser.Select, *sqlparser.Insert, *sqlparser.Update, *sqlparser.OtherRead, *sqlparser.Show, *sqlparser.Explain, *sqlparser.Union:
 		return se.query(ctx, query)
 	case *sqlparser.Delete:
-		ok := se.checkThenDeleteAllRows(ctx, s)
-		if ok {
+		err := se.checkThenDeleteAllRows(ctx, s)
+		if err != nil {
```

Comment:
You have changed the logic here. checkThenDelete is mutually exclusive with passing the query through the engine

Reply:
fixed